cpu/stm32/periph/gpio_all.c

```diff
@@ -72,7 +80,11 @@ static inline GPIO_TypeDef *_port(gpio_t pin)
  */
 static inline int _port_num(gpio_t pin)
 {
+#if defined(CPU_FAM_STM32MP1)
+    return (((pin >> 12) - 2) & 0x0f);
```

Comment:
Not `0xfff` as in `_port`?
And why subtract 2?

Reply:
I substract 2, because into vendor header cpu/stm32/include/vendor/stm32mp157cxx_cm4.h:
```
/*!< MCU_AHB4 */
#define RCC_BASE              (MCU_AHB4_PERIPH_BASE + 0x0000)
#define PWR_BASE              (MCU_AHB4_PERIPH_BASE + 0x1000)
#define GPIOA_BASE            (MCU_AHB4_PERIPH_BASE + 0x2000)
#define GPIOB_BASE            (MCU_AHB4_PERIPH_BASE + 0x3000)
#define GPIOC_BASE            (MCU_AHB4_PERIPH_BASE + 0x4000)
#define GPIOD_BASE            (MCU_AHB4_PERIPH_BASE + 0x5000)
#define GPIOE_BASE            (MCU_AHB4_PERIPH_BASE + 0x6000)
#define GPIOF_BASE            (MCU_AHB4_PERIPH_BASE + 0x7000)
#define GPIOG_BASE            (MCU_AHB4_PERIPH_BASE + 0x8000)
#define GPIOH_BASE            (MCU_AHB4_PERIPH_BASE + 0x9000)
#define GPIOI_BASE            (MCU_AHB4_PERIPH_BASE + 0xA000)
#define GPIOJ_BASE            (MCU_AHB4_PERIPH_BASE + 0xB000)
#define GPIOK_BASE            (MCU_AHB4_PERIPH_BASE + 0xC000)
```
Once you shifted of 12 bits to get port number, there is still an offset of 2 due to RCC_BASE and PWR_BASE, that's the 2 I substract.

it is not 0xFFF because it is a mask to isolate the 4 LSB bits, not to reset to 0 the 12 LSB bits.

Reply:
So why not `(pin - GPIOA_BASE) >> 12`

Reply:
yes ! same result but GPIOA_BASE is more understandable than a 2 coming from space. ;)